commit d2de47e5ceda14aa93551aa47bdd0917f6de7d68
Author: Arjan van de Ven <arjan@linux.intel.com>
Date:   Mon Mar 14 11:06:46 2016 -0600

    UBUNTU: SAUCE: Clear Linux: ksm-wakeups
    
    reduce wakeups in ksm by adding rounding (aligning) when the sleep times are 1 second or longer
    
    Signed-off-by: Arjan van de Ven <arjan@linux.intel.com>
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>

diff --git a/mm/ksm.c b/mm/ksm.c
index 6cb60f46cce5..883a9441e055 100644
--- a/mm/ksm.c
+++ b/mm/ksm.c
@@ -2335,8 +2335,12 @@ static int ksm_scan_thread(void *nothing)
 		try_to_freeze();
 
 		if (ksmd_should_run()) {
-			schedule_timeout_interruptible(
-				msecs_to_jiffies(ksm_thread_sleep_millisecs));
+			if (ksm_thread_sleep_millisecs >= 1000)
+				schedule_timeout_interruptible(
+					msecs_to_jiffies(round_jiffies_relative(ksm_thread_sleep_millisecs)));
+			else
+				schedule_timeout_interruptible(
+					msecs_to_jiffies(ksm_thread_sleep_millisecs));
 		} else {
 			wait_event_freezable(ksm_thread_wait,
 				ksmd_should_run() || kthread_should_stop());
